### DISCUSSION TITLE
EnumToBoolConverter - No property, BindableProperty, or event found for "TrueValues", or mismatching type between value and property.

### DISCUSSION THREAD

**Discussion by BioTurboNick** [CommunityToolkit/Maui#759] [Q&A] [answered]
I believe I followed the documentation correctly, but I'm getting an error:

`No property, BindableProperty, or event found for "TrueValues", or mismatching type between value and property.`

The code:
```
<toolkit:EnumToBoolConverter x:Key="inCountingMode">
    <toolkit:EnumToBoolConverter.TrueValues>
        <e:MeasurementPageMode>Counting</e:MeasurementPageMode>
    </toolkit:EnumToBoolConverter.TrueValues>
</toolkit:EnumToBoolConverter>

public enum MeasurementPageMode
{
    Counting,
    Editing,
    Results
}
```

What did I do wrong?

**Comment by bijington** [comment, marked as answer]
The issue here is that you are assigning a single value into the `TrueValues` property which cannot implicitly be converted to the underlying type of `IList`. To workaround that issue you can add a second value into the `TrueValues` property as below:

```xaml
<toolkit:EnumToBoolConverter x:Key="inCountingMode">
    <toolkit:EnumToBoolConverter.TrueValues>
        <e:MeasurementPageMode>Counting</e:MeasurementPageMode>
        <e:MeasurementPageMode>Counting</e:MeasurementPageMode>
    </toolkit:EnumToBoolConverter.TrueValues>
</toolkit:EnumToBoolConverter>
```

It's not ideal but the compiler will be able to convert that to the desired type. I believe can could create a list in XAML but it would be more verbose than this workaround.

**Reply by danielancines** [reply]
@bijington What's the difference between the converter and the behaviors property from any VisualElement? In this case, we're able to add only one behavior without errors.

**Reply by danielancines** [reply]
I've made some tests and the problem apears to be with Enum type, with strings and int the error don't occur.

**Reply by bijington** [reply]
oh interesting! I suspect this is because `TrueValues` is of type `IList<Enum>` and not the explicit type of `IList< MeasurementPageMode>` from the original example. If that is the case we could possibly look at:

## 1. Allowing the developer to provide the generic type

This would make the types match and possibly solve the issue.

```xaml
<toolkit:EnumToBoolConverter x:Key="inCountingMode" x:TypeArguments="e:MeasurementPageMode">
    <toolkit:EnumToBoolConverter.TrueValues>
        <e:MeasurementPageMode>Counting</e:MeasurementPageMode>
        <e:MeasurementPageMode>Counting</e:MeasurementPageMode>
    </toolkit:EnumToBoolConverter.TrueValues>
</toolkit:EnumToBoolConverter>
```

## 2. Investigate whether a TypeConverter could solve the issue here.

**Reply by danielancines** [reply]
![image](https://user-images.githubusercontent.com/2250055/201555234-d5926770-9282-4b3d-a676-95a7b64bd5c1.png)
I've been running another tests and I notice that, if you try to instanciate a IList of Enum property using List of MeasurementPageMode it'll not work, only if you add the value of MeasurementPageMode, like MeasurementPageMode.Counting. Because of this behavior your workaround works.

**Reply by BioTurboNick** [reply]
It seems like there might be a similar problem with CompareConverter? The docs show you can just put an integer in the XAML, but I'm getting an error. And I see the property in the class is `IComparable?`

**Reply by bijington** [reply]
@BioTurboNick what error are you getting? You should be able to use an integer or double, etc. in XAML as per this:

https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml

**Comment by CliffAgius** [comment]
Closed as answered
